site-manage/site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/rsf/GroupDelProducer.java

```diff
@@ -96,43 +96,59 @@ public void fillComponents(UIContainer tofill, ViewParameters arg1, ComponentChe
+                if (group.isLocked()) {
+                    notDeletable.add(group.getTitle());
+                } else {
+                    String groupId = group.getId();
+                    UIBranchContainer grouprow = UIBranchContainer.make(deleteForm, "group-row:", group.getId());
 
-    			UIOutput.make(grouprow,"group-title",group.getTitle());
-    			
-    			int size = 0;
-    			try
-    			{
-    				size=authzGroupService.getAuthzGroup(group.getReference()).getMembers().size();
-    			}
-    			catch (GroupNotDefinedException e)
-    			{
-    				M_log.debug(this + "fillComponent: cannot find group " + group.getReference());
-    			}
-    			UIOutput.make(grouprow,"group-size",String.valueOf(size));
+					UIOutput.make(grouprow,"group-title",group.getTitle());
 
-    			deletable.add(group.getId());
-				UISelectChoice delete =  UISelectChoice.make(grouprow, "group-select", deleteselect.getFullID(), (deletable.size()-1));
-				delete.decorators = new DecoratorList(new UITooltipDecorator(UIMessage.make("delete_group_tooltip", new String[] {group.getTitle()})));
-				UIMessage message = UIMessage.make(grouprow,"delete-label","delete_group_tooltip", new String[] {group.getTitle()});
-				UILabelTargetDecorator.targetLabel(message,delete);
-				M_log.debug(this + ".fillComponent: this group can be deleted");
-				renderDelete = true;
+					int size = 0;
+					try
+					{
+							size=authzGroupService.getAuthzGroup(group.getReference()).getMembers().size();
+					}
+					catch (GroupNotDefinedException e)
+					{
+							M_log.debug(this + "fillComponent: cannot find group " + group.getReference());
+					}
+					UIOutput.make(grouprow,"group-size",String.valueOf(size));
+
+					deletable.add(group.getId());
+					UISelectChoice delete =  UISelectChoice.make(grouprow, "group-select", deleteselect.getFullID(), (deletable.size()-1));
+					delete.decorators = new DecoratorList(new UITooltipDecorator(UIMessage.make("delete_group_tooltip", new String[] {group.getTitle()})));
+					UIMessage message = UIMessage.make(grouprow,"delete-label","delete_group_tooltip", new String[] {group.getTitle()});
+					UILabelTargetDecorator.targetLabel(message,delete);
+					M_log.debug(this + ".fillComponent: this group can be deleted");
+					renderDelete = true;
+                }
             }
 		}
 
 		deleteselect.optionlist.setValue(deletable.toStringArray());
 		UICommand.make(deleteForm, "delete-groups",  UIMessage.make("editgroup.removegroups"), "#{SiteManageGroupSectionRoleHandler.processDeleteGroups}");
 		UICommand cancel = UICommand.make(deleteForm, "cancel", UIMessage.make("editgroup.cancel"), "#{SiteManageGroupSectionRoleHandler.processCancelDelete}");
 		cancel.parameters.add(new UIDeletionBinding("#{destroyScope.resultScope}"));
-   
+
+		if (!notDeletable.isEmpty()) {
+			String groupsTitles = "";
+
+			for (Object groupTitle : notDeletable) {
+				groupsTitles += groupTitle.toString() + " ";
```

Comment:
I suggest to use StringJoiner (https://docs.oracle.com/javase/8/docs/api/java/util/StringJoiner.html) and use comma as separator:

GR1,GR2,GR3 ....